tests/test_de.rs

```diff
@@ -242,6 +243,48 @@ fn test_enum_representations() {
     test_de_no_value(yaml, &expected);
 }
 
+#[test]
+fn test_enum_outer_flatten() {
+    #[derive(Deserialize, PartialEq, Debug)]
+    enum Enum {
+        A,
+        B,
+        C(String),
+    }
+    #[derive(Deserialize, PartialEq, Debug)]
+    struct Inner {
+        a: Enum,
+    }
+    #[derive(Deserialize, PartialEq, Debug)]
+    struct Outer {
+        #[serde(flatten)]
+        inner: Inner,
+    }
+
+    let yaml: &str = indoc! {"
+        a: !C x
+    "};
+    let inner = Inner {
+        a: Enum::C("x".to_owned()),
+    };
+    test_de(yaml, &inner);
+    test_de(yaml, &Outer { inner });
+    let yaml: &str = indoc! {"
+        a: !C
+    "};
+    // let expected = Outer {
+    //     inner: Inner {
+    //         a: Enum::C(String::new()),
+    //     },
+    // };
+    // test_de_no_value(yaml, &expected);
+    // This should fail. Blocked by by https://github.com/serde-rs/serde/issues/1183
+    assert!(serde_yaml_ng::from_str::<Outer>(yaml).is_err());
```

Comment:
Error is expected here, because according to the description in the review, this should be deserialized to `None` if it would be `Option`:
```suggestion
    let yaml: &str = indoc! {"
        a: !C
    "};
    assert!(serde_yaml_ng::from_str::<Inner>(yaml).is_err());
    assert!(serde_yaml_ng::from_str::<Outer>(yaml).is_err());
```
Of course, if deserialization of `Inner` works (I didn't check that), keep that comment.